### PR TITLE
refactor(fieldset): convert tests to RTL

### PR DESCRIPTION
**Pull request by nineteen88**
### Proposed behaviour

- tidy up existing tests and convert to RTL
- move the required and isOptional tests to Chromatic and put a placeholder test since we can't test it in RTL but need the coverage

### Current behaviour

Tests currently are in Enzyme and there are some styling tests that no longer are required

### Checklist

- [x] Commits follow our style guide
- [x] Unit tests added or updated if required
- [x] Storybook added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

**Review by DipperTheDan — APPROVED** [review]

File: src/components/fieldset/fieldset-test.stories.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import Textbox from "../textbox";
 
 export default {
   title: "Fieldset/Test",
-  includeStories: ["Default"],
+  includeStories: ["Default", "OptionalFieldset"],
```
Comment (tomdavies73) [inline src/components/fieldset/fieldset-test.stories.tsx:7]: **question** (non-blocking): was the required fieldset story meant to be included here too? No worries if not

File: src/components/fieldset/fieldset.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Fieldset from "./fieldset.component";
+import Textbox from "../textbox";
+import "@testing-library/jest-dom/extend-expect";
+
+test("Fieldset Legend is rendered if supplied", () => {
+  render(<Fieldset legend="Legend" />);
+  expect(screen.getByText("Legend")).toBeInTheDocument();
```
Comment (tomdavies73) [inline src/components/fieldset/fieldset.test.tsx:9]: **suggestion**: `toBeVisible` could be better here

File: src/components/fieldset/fieldset-test.stories.tsx (hunk 1 above)
Comment (nineteen88) [reply src/components/fieldset/fieldset-test.stories.tsx:7]: Yes it should be

**Review by tomdavies73 — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**